omnigibson/utils/teleop_utils.py

```diff
@@ -0,0 +1,540 @@
+                XRCoreEventType.hand_joints, self.get_hand_tracking_data, name="hand tracking"
+            )
+
+    def xr2og(self, transform: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Apply the orientation offset from the Omniverse XR coordinate system to the OmniGibson coordinate system
+        Args:
+            transform (np.ndarray): the transform matrix in the Omniverse XR coordinate system
+        Returns:
+            tuple(np.ndarray, np.ndarray): the position and orientation in the OmniGibson coordinate system
+        """
+        pos, orn = T.mat2pose(np.array(transform).T)
+        orn = T.quat_multiply(orn, np.array([0.5, -0.5, -0.5, -0.5]))
+        return pos, orn
+    
+    def og2xr(self, pos: np.ndarray, orn: np.ndarray) -> np.ndarray:
+        """
+        Apply the orientation offset from the OmniGibson coordinate system to the Omniverse XR coordinate system
+        Args:
+            pos (np.ndarray): the position in the OmniGibson coordinate system
+            orn (np.ndarray): the orientation in the OmniGibson coordinate system
+        Returns:
+            np.ndarray: the transform matrix in the Omniverse XR coordinate system
+        """
+        orn = T.quat_multiply(np.array([-0.5, 0.5, 0.5, -0.5]), orn)
+        return T.pose2mat((pos, orn)).T.astype(np.float64)
+
+    @property
+    def is_enabled(self) -> bool:
+        """
+        Checks whether the VR system is enabled
+        Returns:
+            bool: whether the VR system is enabled
+        """
+        return self.vr_profile.is_enabled()
+
+    def start(self) -> None:
+        """
+        Enabling the VR profile
+        """
+        self.vr_profile.request_enable_profile()
+        for _ in range(50):
+            og.sim.step()
+        assert self.vr_profile.is_enabled(), "[VRSys] VR profile not enabled!"
+        self._update_devices()
+
+    def stop(self) -> None:
+        """
+        disable VR profile
+        """
+        self.xr_core.request_disable_profile()
+        self.sim.step()
+        assert not self.vr_profile.is_enabled(), "[VRSys] VR profile not disabled!"
+
+    def update(self):
+        """
+        Steps the VR system
+        """
+        # Optional move anchor
+        if self.enable_touchpad_movement:
+            self.move_anchor(pos_offset=self.compute_anchor_offset_with_controller_input())
+        if self.align_anchor_to_robot_base:
+            robot_base_pos, robot_base_orn = self.robot.get_position_orientation()
+            self.vr_profile.set_virtual_world_anchor_transform(self.og2xr(robot_base_pos, robot_base_orn[[0, 2, 1, 3]]))
+        # update raw data
+        self._update_devices()
+        self._update_device_transforms()
+        self._update_button_data()
+        if self.use_hand_tracking:
+            self._update_hand_tracking_data()
+        # generate teleop data
+        self.teleop_data["transforms"]["base"] = np.zeros(4)
+        if 1 in self.controllers:
```

Comment:
what is `1` and `0`?

Reply:
This is the index of the device from the Omniverse XR plugin, 1 is right controller, 0 is left controller, and 2+ is other controller

Reply:
changed to `left` and `right` for better readability